worker.go

```diff
@@ -43,17 +44,49 @@ func NewWorkerMetrics() WorkerMetrics {
 	}
 }
 
+// Create an entry in wm for the given metrickey, if it does not already exist.
+// Returns true if the metric entry was created and false otherwise.
+func (wm WorkerMetrics) Upsert(mk MetricKey, tags []string) bool {
```

Comment:
you are correct that nothing uses the bool right now, but returning the created sampler object is a static typing nightmare. The sample method does not have the same signature across all four types (sets consume strings), and not all samplers implement a combine method. It's going to be a _ton_ of type assertion. You can't get rid of the switch in ProcessMetric/ImportMetric either, because you still need to switch over something to determine what type it is. I tossed the idea around briefly but I don't see any concrete code wins from doing it, and the samplers are structurally different enough that I do not see value in squeezing them into a uniform interface.